usr/sbin/registercloudguest

```diff
@@ -128,58 +129,94 @@ def register_modules(extensions, products, registered=[]):
-# Support custom config file with -f command line option
-config_file = None
-
-if '-f' in sys.argv:
-    idx = sys.argv.index('-f')
-    config_file = sys.argv[idx+1]
-
-force_new_registration = False
-if '--force-new' in sys.argv:
-    force_new_registration = True
-
-if '--delay' in sys.argv:
-    arg_index = sys.argv.index('--delay')
-    delay_time = sys.argv[arg_index + 1]
-    time.sleep(int(delay_time))
-
-user_smt_ip = None
-if '--smt-ip' in sys.argv:
-    arg_index = sys.argv.index('--smt-ip')
-    user_smt_ip = sys.argv[arg_index + 1]
-    # Validation has to wait for Python 3 implementation
-
-user_smt_fqdn = None
-if '--smt-fqdn' in sys.argv:
-    arg_index = sys.argv.index('--smt-fqdn')
-    user_smt_fqdn = sys.argv[arg_index + 1]
-
-user_smt_fp = None
-if '--smt-fp' in sys.argv:
-    arg_index = sys.argv.index('--smt-fp')
-    user_smt_fp = sys.argv[arg_index + 1]
-
-if user_smt_ip or user_smt_fqdn or user_smt_fp:
-    if not (user_smt_ip and user_smt_fqdn and user_smt_fp):
+argparse = argparse.ArgumentParser(description='Register on-demand instance')
+argparse.add_argument(
+    '--clean',
+    action='store_true',
+    dest='clean_up',
+    default=False,
+    help='Clean up registration data'
+)
+argparse.add_argument(
+    '-d', '--delay',
+    default=0,
+    dest='delay_time',
+    help='Delay the start of registartion by given value in seconds',
+    type=int
+)
+# default config file location not set it is encoded in the utils function
+# get_config(()
+argparse.add_argument(
+    '-f', '--config-file',
+    dest='config_file',
+    help='Path to config file, default: /etc/regionserverclnt.cfg',
+)
+argparse.add_argument(
+    '--force-new',
+    action='store_true',
+    dest='force_new_registration',
+    default=False,
+    help='Force a new registration, for exceptional cases only',
+)
+help_msg='The update server cert fingerprint. Use '
+help_msg+='in exceptional cases only'
+argparse.add_argument(
+    '--smt-fp',
+    dest='user_smt_fp',
+    help=help_msg
+)
+help_msg='Use a specific FQDN to connect to the update server, use '
+help_msg+='in exceptional cases only'
+argparse.add_argument(
+    '--smt-fqdn',
+    dest='user_smt_fqdn',
+    help=help_msg
+)
+help_msg='Use a specific IP to connect to the update server, use '
+help_msg+='in exceptional cases only'
+argparse.add_argument(
+    '--smt-ip',
+    dest='user_smt_ip',
+    help=help_msg
+)
+
+args = argparse.parse_args()
+
+if args.user_smt_ip or args.user_smt_fqdn or args.user_smt_fp:
+    if not (args.user_smt_ip and args.user_smt_fqdn and args.user_smt_fp):
         msg = '--smt-ip, --smt-fqdn, and --smt-fp must be used together'
         print(msg, file=sys.stderr)
+        sys.exit(1)
+
+if args.clean_up and args.force_new_registration:
+    msg = '--clean and --force-new are incompatible, use one or the other'
+    print(msg, file=sys.stderr)
+    sys.exit(1)
+
+time.sleep(int(args.delay_time))
 
+if args.clean_up:
+    utils.remove_registration_data()
```

Comment:
1. I would suggest calling `utils.clean_smt_cache()` here (if the intended use is to create a snapshot of an instance, then there's no point to bring along SMT cache), `--force-new` calls it too:

https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/a938dc18613c1f82f626c9f2376ecd6c686b9bfb/usr/sbin/registercloudguest#L238-L240

2. And maybe even throw in a call to `utils.clear_new_registration_flag()`? (Otherwise the marker would be cleaned up next time `registercloudguest` is called, but it still might be a good idea to clean up everything)